src/main/java/com/amazon/opendistroforelasticsearch/sql/antlr/semantic/types/function/ScalarFunction.java

```diff
@@ -91,7 +91,15 @@
         func(T(STRING)).to(T),
         func(T(STRING), STRING).to(T)
     ),
-    YEAR(func(DATE).to(INTEGER));
+    YEAR(func(ESDataType.DATE).to(INTEGER)),
+    MONTH(func(ESDataType.DATE).to(INTEGER)),
+    MONTHNAME(func(ESDataType.DATE).to(STRING)),
+    DAYOFMONTH(func(ESDataType.DATE).to(INTEGER)),
+    DATE(func(ESDataType.DATE).to(ESDataType.DATE)),
+    TIMESTAMP(func(ESDataType.DATE).to(ESDataType.DATE)),
+    MAKETIME(func(INTEGER, INTEGER, INTEGER).to(ESDataType.DATE)),
+    NOW(),
+    CURDATE();
```

Comment:
Should be `() -> DATE` ? None specification means it's unknown and we want to bypass any type check for now.

Reply:
Done. Thanks!